rdbms/src/main/java/io/spine/server/storage/jdbc/query/Parameters.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.storage.jdbc.query;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Maps.newHashMap;
+import static java.util.Collections.disjoint;
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * A {@linkplain Parameter query parameters} with identifiers.
+ *
+ * <p>{@linkplain #getIdentifiers() Identifiers} allows to distinguish a place
+ * in a query where a parameter should be inserted.
+ *
+ * <p>This class allows to encapsulate setting of parameters to a concrete implementation.
+ *
+ * @author Dmytro Grankin
+ */
+public class Parameters {
+
+    private final Map<Integer, Parameter> parameters;
+
+    private Parameters(Map<Integer, Parameter> parameters) {
+        this.parameters = parameters;
+    }
+
+    /**
+     * Obtains a set of parameter identifiers.
+     *
+     * @return parameter identifiers
+     */
+    public Set<Integer> getIdentifiers() {
+        return parameters.keySet();
```

Comment:
That looks suspicious. I wouldn't return a collection like this.

Reply:
Done.